### DISCUSSION TITLE
[RFC] Mutex lock / prevent simultaneous executions

### DISCUSSION THREAD

**Discussion by elado** [vercel/turborepo#7256] [Ideas]
### Describe the feature you'd like to request

Prevent the ability to execute `turbo` on the same package/script in two terminals.

Use case:
I have multiple web servers that are different packages, and they have some shared deps.
Running `yarn turbo run start --filter=app1` in one terminal and `yarn turbo run start --filter=app2` in another results in dependencies building twice. When app2 builds a shared dep it's also cleaning the dist folder before, causing app1 to break.

### Describe the solution you'd like

A file system based locking mechanism on a package/script pair.
Before executing actual script in a package, check that it isn't currently running. If it is, wait for the other process to finish first.
Clean locks in case of errors.

### Describe alternatives you've considered

A wrapper around turbo execution with mutex/lock file (e.g. with [mutex](https://github.com/isaacs/slocket) [packages](https://www.npmjs.com/package/@ster5/global-mutex)). This works only when the scripts executed are finite and not continuous like starting a web server, otherwise it stays locked forever.

**Comment by jaredpalmer** [comment]
Great idea. We are architecting a daemon (`turbod`) and so shared watch mode / mutex locks is something that it would take care of elegantly. Going to add this to our design constraints.

**Comment by elado** [comment]
Wondering if this was ever integrated into the daemon? I still see double builds happening when running in parallel.

**Comment by RobinVdBroeck** [comment]
Any updates on this? Is it still being planned?
